### ISSUE TITLE
Cannot get it to pair with homey pro 2019 and dirigera.

### ISSUE THREAD

**Issue by sigve888** [theeuwes-it/homey-dirigera-gateway#1]

**Comment by theeuwes-it** [comment]
Hi @sigve888 ,
Could you provide me with a description of what you have tried?
Please make sure your Dirigera hub is updated to the latest version (System settings -> <Your Dirigera> -> Hub and app version) before trying to connect your Dirigera to your Homey

**Comment by sigve888** [comment]
> Hi @sigve888 ,
> 
> Could you provide me with a description of what you have tried?
> 
> Please make sure your Dirigera hub is updated to the latest version (System settings -> <Your Dirigera> -> Hub and app version) before trying to connect your Dirigera to your Homey


Hi, yes! Wasn't sure I would get a reply. Let me dump some more info!
Sent a diagnostic report with ref: c8ac0fa6-f767-4d7b-bd31-9804eb442155

Hub
Model: DIRIGERA Hub for smart products
Serial: x-x-x-x-xxxe89fe64
Produsent: IKEA of Sweden
Firmware
2.685.0
App
1.40.0#1182

Getting error when trying to add my blinds: 'Timeout after 30000 ms'.

**Comment by theeuwes-it** [comment]
I received your report. It looks like the app was trying to get the Access Token, but it didn't receive it in time. Did you press the action button (on the bottom) of the Dirigera after clicking the "Authenticate" button?
If so, could you please try the flow again? So, click "Authenticate", click the action button and then close the popup?

**Comment by sigve888** [comment]
> I received your report. It looks like the app was trying to get the Access Token, but it didn't receive it in time. Did you press the action button (on the bottom) of the Dirigera after clicking the "Authenticate" button?
> 
> If so, could you please try the flow again? So, click "Authenticate", click the action button and then close the popup?

Yes. Give me half an hour or something. I'll publish a new diagnostic report.

**Comment by sigve888** [comment]
Reinstalled the app and did a power cycle on both homey and dirigera. Authenticated , pressee button, closed the popup and saved . Sane error (timed out after 30000 ms).

Diagnostic: a7922acf-bf71-4e3f-9108-97472fe09322

**Comment by dsvoren** [comment]
I second this. I get the same error "timed out after 30000 ms" trying to add bulbs and sockets.

I have latest the firmware (2.685.0) on Dirigera and HP2023.

The authentication token has been set by means of the Action button. 

Diagnostics: 6486926d-b940-44b9-be4a-7518949b1335

I thing I could think of is that I have a mesh network (Deco), could that affect network discovery to the Dirigera hub?

**Comment by sigve888** [comment]
Fyi the blinds I'm using is TREDANSEN.

**Comment by theeuwes-it** [comment]
Thanks for you replies! It will be some time before I can check this out. I'll let you know when I know more

**Comment by theeuwes-it** [comment]
Hi guys,

My latest test build improves the authentication process so the timeouts should occur less (hopefully not anymore at all).
Please try it out (https://homey.app/a/com.theeuwes-it.ikea.dirigera/test) and let me know!

**Comment by godkatt** [comment]
> Hi guys,
> 
> My latest test build improves the authentication process so the timeouts should occur less (hopefully not anymore at all). Please try it out (https://homey.app/a/com.theeuwes-it.ikea.dirigera/test) and let me know!

Hey! Indeed, this new experimental version (1.0.9) did the trick. The bulbs and outlets were added (fyi much faster than the time out limit of 30s). Now it works like a charm, awesome app. Thanks for looking into this!
